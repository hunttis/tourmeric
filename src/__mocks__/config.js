export const config = {
  default: {
    titleText: 'Foo',
    apiKey: 'Foo',
    authDomain: 'Foo',
    databaseURL: 'Foo',
    projectId: 'Foo',
    storageBucket: 'Foo',
    messagingSenderId: 'Foo',
    fileMetadataFactory: (uploadRes) => {
      const { metadata: { name, fullPath, downloadURLs } } = uploadRes;
      return {
        name,
        fullPath,
        downloadURL: downloadURLs,
      };
    },
  },
};

exports.get = function get() {
  return config.default;
};
