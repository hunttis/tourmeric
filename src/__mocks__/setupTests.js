import { configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16'; // eslint-disable-line

configure({ adapter: new Adapter() });
