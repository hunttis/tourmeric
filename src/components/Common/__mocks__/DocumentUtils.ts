
export default function scrollIntoview(_elementId: string) {
  console.log('scrolling');
}

export function setListener(_eventType: string, _callback: () => {}) {
  console.log('adding listener');
}

export function removeClassFromHtml(className: string) {
  console.log('removing ', className);
}

export function addClassToHtml(className: string) {
  console.log('adding ', className);
}