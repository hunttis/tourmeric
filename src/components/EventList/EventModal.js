import React, { Component, Fragment } from 'react';
import { Translate } from 'react-localize-redux';
import PropTypes from 'prop-types';
import _ from 'lodash';
import Moment from 'react-moment';
import { isLoaded } from 'react-redux-firebase';

export default class EventModal extends Component {

  foo() {

  }

  render() {
    const { eventId, events, settings, closeModal, participations, categories } = this.props;

    if (isLoaded(events) && isLoaded(participations) && isLoaded(settings)) {

      const eventContent = _.get(events, eventId, {});
      const category = _.get(categories, eventContent.category);
      const dateFormat = _.get(settings, 'dateFormat');
      let participationsForEvent = Object.values(_.get(participations, eventId, []));
      participationsForEvent = _.sortBy(participationsForEvent, ['date']);

      return (
        <div className="modal is-active" id={`modal${eventId}`} key={`modal${eventId}`}>
          <div className="modal-background" onClick={closeModal} />
          <div className="modal-content">
            <div className="box is-rounded">
              <div className="columns is-multiline">

                <div className="column is-9">
                  <h1 className="title has-text-success">{eventContent.name}</h1>
                </div>

                {category &&
                <div className="column is-3 has-text-right">
                  <figure className="image is-64x64 is-pulled-right">
                    <img alt="" src={category.image} />
                  </figure>
                </div>
            }

                <div className="column is-12">
                  {eventContent.date &&
                  <Fragment>
                    <i className="fas fa-calendar" />&nbsp;&nbsp;
                    <Moment format={dateFormat}>{eventContent.date}</Moment>
                    <br />
                  </Fragment>
              }

                  {eventContent.time &&
                  <Fragment><i className="fas fa-clock" />&nbsp;&nbsp;{eventContent.time}<br /></Fragment>
              }

                  {eventContent.format &&
                  <Fragment><i className="fas fa-book" />&nbsp;&nbsp;{eventContent.format}<br /></Fragment>
              }

                  {eventContent.rulesLevel &&
                  <Fragment><i className="fas fa-balance-scale" />&nbsp;&nbsp;{eventContent.rulesLevel}<br /></Fragment>
              }

                  {eventContent.entryFee &&
                  <Fragment><i className="fas fa-money-bill-alt" />&nbsp;&nbsp;{eventContent.entryFee}&nbsp;â‚¬<br /></Fragment>
              }
                </div>

                <ModalItem translationKey="notes" content={eventContent.notes} />
                <ModalItem translationKey="prizes" content={eventContent.prizes} />

                {eventContent.link &&
                <Fragment>
                  <div className="column is-12">
                    <div className="subtitle has-text-info"><Translate id="link" /></div>
                  </div>

                  <div className="column is-1" />
                  <div className="column is-11">
                    <p>
                      <a href={eventContent.link}>{eventContent.link}</a>
                    </p>
                  </div>
                </Fragment>
            }
                {!_.isEmpty(participations) &&
                <Fragment>

                  <div className="column is-12">
                    <div className="subtitle has-text-info"><Translate id="participants" /> ({participationsForEvent.length} / {eventContent.playerSlots})</div>
                  </div>
                  <div className="column is-1" />
                  <div className="column is-11">
                    <ParticipantList participations={participationsForEvent} />
                  </div>
                </Fragment>
            }
              </div>
            </div>
          </div>
          <button className="modal-close is-large" aria-label="close" onClick={closeModal} />
        </div>
      );
    }
    return <div />;
  }
}

const ModalItem = ({ translationKey, content }) => (
  <Fragment>
    {content &&
    <Fragment>
      <div className="column is-12">
        <div className="subtitle has-text-info"><Translate id={translationKey} /></div>
      </div>

      <div className="column is-1" />
      <div className="column is-11">
        <p>
          {content}
        </p>
      </div>
    </Fragment>
  }
  </Fragment>
);

const ParticipantList = ({ participations }) => (
  <div>
    {participations && participations.map((participation, index) => {
      const coloration = index % 2 === 0 ? 'has-background-black has-text-white' : 'has-background-white has-text-black';
      return (
        <div className="columns" key={`participantModal-${participation.userId}`}>
          <div className="column is-3 is-mobile has-text-right is-fixed-bottom commenter">{index + 1}. {participation.firstName} {participation.lastName}</div>
          {participation.comment &&
          <Fragment>
            <div className="column is-9 is-mobile is-paddingless">
              <div className={`speech-bubble ${coloration}`}>
                {participation.comment}
              </div>
            </div>
          </Fragment>
        }
        </div>
      );
    })}
  </div>
);

EventModal.propTypes = {
  eventId: PropTypes.string,
  events: PropTypes.object,
  closeModal: PropTypes.func,
  participations: PropTypes.object,
  categories: PropTypes.object,
  settings: PropTypes.object,
};

ParticipantList.propTypes = {
  participations: PropTypes.array,
};

ModalItem.propTypes = {
  translationKey: PropTypes.string,
  content: PropTypes.string,
};
