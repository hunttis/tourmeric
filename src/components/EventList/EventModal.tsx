import React from 'react';
import { Translate } from 'react-localize-redux';
import _ from 'lodash';
import Moment from 'react-moment';
import { isLoaded } from 'react-redux-firebase';
import { ModalItem } from './ModalItem';
import { ParticipantList } from './ParticipantList';
import { TourmericEvent } from '~/models/Events';
import { Participation } from '~/models/ReduxState';
import { Category } from '~/models/Category';
import { Settings } from '~/models/Settings';

interface Props {
  eventId: string;
  events: { [key: string]: TourmericEvent };
  closeModal: () => void;
  participations: { [key: string]: Participation };
  categories: { [key: string]: Category };
  settings: Settings;
}

export const EventModal = ({
  eventId,
  events,
  settings,
  closeModal,
  participations,
  categories,
}: Props) => {
  if (isLoaded(events) && isLoaded(participations) && isLoaded(settings)) {
    const eventContent = _.get(events, eventId);

    if (!eventContent) {
      return (
        <div>
          <Translate id="noevent" />
        </div>
      );
    }

    const category = _.get(categories, eventContent.category);
    const dateFormat = _.get(settings, 'dateFormat');
    let participationsForEvent = Object.values(
      _.get(participations, eventId, []),
    );
    participationsForEvent = _.sortBy(participationsForEvent, ['date']);

    return (
      <div
        className="modal is-active"
        id={`modal${eventId}`}
        key={`modal${eventId}`}
      >
        <div className="modal-background" onClick={closeModal} />
        <div className="modal-content">
          <div className="box is-rounded">
            <div className="columns is-multiline">
              <div className="column is-9">
                <h1 className="title has-text-success">{eventContent.name}</h1>
              </div>

              {category && (
                <div className="column is-3 has-text-right">
                  <figure className="image is-64x64 is-pulled-right">
                    <img alt="" src={category.image} />
                  </figure>
                </div>
              )}

              <div className="column is-12">
                {eventContent.date && (
                  <>
                    <i className="fas fa-calendar" />
                    &nbsp;&nbsp;
                    <Moment format={dateFormat}>{eventContent.date}</Moment>
                    <br />
                  </>
                )}

                {eventContent.time && (
                  <>
                    <i className="fas fa-clock" />
                    &nbsp;&nbsp;{eventContent.time}
                    <br />
                  </>
                )}

                {eventContent.format && (
                  <>
                    <i className="fas fa-book" />
                    &nbsp;&nbsp;{eventContent.format}
                    <br />
                  </>
                )}

                {eventContent.rulesLevel && (
                  <>
                    <i className="fas fa-balance-scale" />
                    &nbsp;&nbsp;{eventContent.rulesLevel}
                    <br />
                  </>
                )}

                {eventContent.entryFee && (
                  <>
                    <i className="fas fa-money-bill-alt" />
                    &nbsp;&nbsp;{eventContent.entryFee}
                    <br />
                  </>
                )}
              </div>

              {eventContent.notes && (
                <ModalItem
                  translationKey="notes"
                  content={eventContent.notes}
                />
              )}
              {eventContent.prizes && (
                <ModalItem
                  translationKey="prizes"
                  content={eventContent.prizes}
                />
              )}

              {eventContent.link && (
                <>
                  <div className="column is-12">
                    <div className="subtitle has-text-info">
                      <Translate id="link" />
                    </div>
                  </div>

                  <div className="column is-1" />
                  <div className="column is-11">
                    <p>
                      <a
                        href={eventContent.link}
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        {eventContent.link}
                      </a>
                    </p>
                  </div>
                </>
              )}

              {!_.isEmpty(participations) && (
                <>
                  <div className="column is-12">
                    <div className="subtitle has-text-info">
                      <Translate id="participants" /> (
                      {participationsForEvent.length} /{' '}
                      {eventContent.playerSlots})
                    </div>
                  </div>
                  <div className="column is-1" />
                  <div className="column is-11">
                    <ParticipantList
                      participations={participationsForEvent}
                      maxParticipants={parseInt(
                        _.get(eventContent, 'playerSlots', '0'),
                        10,
                      )}
                    />
                  </div>
                </>
              )}
            </div>
          </div>
        </div>
        <button
          className="modal-close is-large"
          aria-label="close"
          onClick={closeModal}
        />
      </div>
    );
  }
  return <div />;
};
