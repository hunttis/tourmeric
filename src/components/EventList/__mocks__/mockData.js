export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
