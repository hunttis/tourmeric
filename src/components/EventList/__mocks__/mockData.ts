import { FirebaseProfile, Participation, User, ProviderData, Location } from '~/models/ReduxState';
import { TourmericEvent } from '~/models/Events';
import { Settings } from '~/models/Settings';
import { UploadedFile, Category } from '~/models/Category';

export const mockEventId = 'mockEvent1';

export const mockUserId = 'mockUserId';

export const mockLocation: Location = {
  pathname: '/events/2018/10',
  search: 'search',
  hash: 'hash',
  key: 'key',
};

export const mockProviderData: ProviderData = {
  displayName: 'MockName',
  email: 'mock@userton.com',
  photoURL: 'http://url',
  providerId: 'facebook',
  uid: '1234567890',
};

export const mockUser: User = {
  avatarUrl: 'http://www.google.fi/google.png',
  displayName: 'Mock Userton',
  firstName: 'Mock',
  lastName: 'Userton',
  providerData: mockProviderData,
  email: 'mock@userton.com',
  username: 'Mockie',
  dciNumber: '123456',
  role: 'user',
  active: true,
};

export const mockProfile: FirebaseProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{
    displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890',
  }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent: TourmericEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
  eventType: 'single',
};

export const mockUnorderedEvents: {[key: string]: TourmericEvent} = {
  mockEvent1: mockSingleEvent,
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents: {key: string, value: TourmericEvent}[] = [
  { key: 'mockEvent1',
    value: mockSingleEvent },
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings: Settings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos: {[key: string]: UploadedFile} =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png', name: 'image' },
  };

export const mockParticipations: {[key: string]: Participation} = {
  mockEventId1: {
    mockUserId: {
      comment: 'Jee',
      date: '2018-06-23T09:40:26.992Z',
      firstName: 'Test',
      lastName: 'Person',
      userId: 'mockUserId',
    },
  },
};

export const mockCategories: {[key: string]: Category} = {
  category1: {
    abbreviation: 'WH40K',
    logo: '1527227370219137123',
    name: 'Warhammer 40,000',
    type: 'Miniatures',
  },
  category2: {
    abbreviation: 'MtG',
    logo: '152722965771313789',
    name: 'Magic: The Gathering',
    type: 'TCG',
  },
  '-LAlvCmIWNzYKpa7-7HI': {
    abbreviation: 'Lautapelit',
    logo: '152719272536668394',
    name: 'Lautapelit / Board games',
    type: 'Boardgame',
  },
  '-LCmESny8m9JVEFQ_qOt': {
    abbreviation: 'PKM',
    logo: '152722747904159438',
    name: 'Pok√©mon',
    type: 'TCG',
  },
  '-LDImTAGxkzrorL0tQdi': {
    abbreviation: 'VG',
    logo: '152719372155055672',
    name: 'Videogames',
    type: 'Videogames',
  },
  '-LDIw-1LFgiM0EJZGfus': {
    abbreviation: 'Necromunda',
    logo: '15271953483568546',
    name: 'Necromunda',
    type: 'Miniatures',
  },
};
