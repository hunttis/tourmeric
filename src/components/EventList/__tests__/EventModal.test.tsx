import React from 'react';
import { shallow } from 'enzyme';
import { EventModal } from '../EventModal';
import { mockEventId, mockParticipations, mockUnorderedEvents, mockCategories, mockSettings } from '../__mocks__/mockData';
import '../../../__mocks__/setupTests';

describe('EventModal tests', () => {

  it('Compares EventModal to snapshot', () => {

    const closeModalFunction = () => {};

    const modal = shallow(<EventModal
      eventId={mockEventId}
      events={mockUnorderedEvents}
      settings={mockSettings}
      closeModal={closeModalFunction}
      participations={mockParticipations}
      categories={mockCategories}
    />);

    expect(modal).toMatchSnapshot();
  });
});
