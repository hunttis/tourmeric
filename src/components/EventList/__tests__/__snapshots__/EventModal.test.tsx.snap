// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventModal tests Compares EventModal to snapshot 1`] = `
<div
  className="modal is-active"
  id="modalmockEvent1"
  key="modalmockEvent1"
>
  <div
    className="modal-background"
    onClick={[Function]}
  />
  <div
    className="modal-content"
  >
    <div
      className="box is-rounded"
    >
      <div
        className="columns is-multiline"
      >
        <div
          className="column is-9"
        >
          <h1
            className="title has-text-success"
          >
            Warhammer thing
          </h1>
        </div>
        <div
          className="column is-3 has-text-right"
        >
          <figure
            className="image is-64x64 is-pulled-right"
          >
            <img
              alt=""
            />
          </figure>
        </div>
        <div
          className="column is-12"
        >
          <i
            className="fas fa-calendar"
          />
            
          <t
            ago={false}
            calendar={false}
            decimal={false}
            element="time"
            filter={[Function]}
            format="DD.MM.YYYY"
            fromNow={false}
            interval={60000}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={false}
            utc={false}
            withTitle={false}
          >
            2018-09-24
          </t>
          <br />
          <i
            className="fas fa-clock"
          />
            
          Kill everything
          <br />
          <i
            className="fas fa-book"
          />
            
          Miniatures
          <br />
          <i
            className="fas fa-balance-scale"
          />
            
          Casual
          <br />
          <i
            className="fas fa-money-bill-alt"
          />
            
          10
          <br />
        </div>
        <Component
          content="Bring your war face!"
          translationKey="notes"
        />
        <Component
          content="Glory!"
          translationKey="prizes"
        />
        <div
          className="column is-12"
        >
          <div
            className="subtitle has-text-info"
          >
            <Translate
              id="link"
            />
          </div>
        </div>
        <div
          className="column is-1"
        />
        <div
          className="column is-11"
        >
          <p>
            <a
              href="http://www.google.fi"
              rel="noopener noreferrer"
              target="_blank"
            >
              http://www.google.fi
            </a>
          </p>
        </div>
        <div
          className="column is-12"
        >
          <div
            className="subtitle has-text-info"
          >
            <Translate
              id="participants"
            />
             (
            0
             /
             
            10
            )
          </div>
        </div>
        <div
          className="column is-1"
        />
        <div
          className="column is-11"
        >
          <Component
            maxParticipants={10}
            participations={Array []}
          />
        </div>
      </div>
    </div>
  </div>
  <button
    aria-label="close"
    className="modal-close is-large"
    onClick={[Function]}
  />
</div>
`;
